tm2/pkg/sdk/params/keeper.go

```diff
@@ -116,6 +119,14 @@ func (pk ParamsKeeper) SetBytes(ctx sdk.Context, key string, value []byte) {
 }
 
 func (pk ParamsKeeper) getIfExists(ctx sdk.Context, key string, ptr interface{}) {
+	if bz, ok := pk.cache.Get(key); ok {
+		if bz == nil {
+			return
+		}
```

Comment:
what is the meaning of that code? can we have nil values in our cache? why?

Reply:
I was considering the two situations, and I feel that since the goal is to avoid unnecessary lookups, caching the "not found" error prevents a misconfigured script from periodically querying the same nonexistent value for a real lookup. 

It's just a feeling; I'm not sure if that's the best approach.